core/drivers/crypto/se050/core/huk.c

```diff
@@ -16,15 +16,12 @@ int tee_otp_get_die_id(uint8_t *buffer, size_t len)
 	size_t se050_huk_len = sizeof(se050_huk);
 	sss_status_t status = kStatus_SSS_Fail;
 
-	memset(buffer, 0, len);
-
 	status = sss_se05x_session_prop_get_au8(se050_session,
 						kSSS_SessionProp_UID,
 						se050_huk, &se050_huk_len);
 	if (status != kStatus_SSS_Success)
 		return -1;
 
-	memcpy(buffer, se050_huk, MIN(len, se050_huk_len));
-
-	return 0;
+	return tee_hash_createdigest(TEE_ALG_SHA256, se050_huk, se050_huk_len,
```

Comment:
`tee_otp_get_die_id()` is supposed to return `-1` on error and `0` on success. `tee_hash_createdigest()` instead returns a `TEE_Result` which will be `0` on success, but not `-1` on error. I'd prefer if `tee_otp_get_die_id()` also did return a `TEE_Result`, but that's not the case and not really in scope of this PR either.

Please fix so `-1` is returned in case of error.

Reply:
@jenswi-linaro good catch :)